src/pybind/mgr/cephadm/services/nvmeof.py

```diff
@@ -69,6 +70,17 @@ def prepare_create(self, daemon_spec: CephadmDaemonDeploySpec) -> CephadmDaemonD
         self.mgr.log.info(f"gateway address: {addr} from {map_addr=} {spec.addr=} {host_ip=}")
         discovery_addr = map_discovery_addr or spec.discovery_addr or host_ip
         self.mgr.log.info(f"discovery address: {discovery_addr} from {map_discovery_addr=} {spec.discovery_addr=} {host_ip=}")
+        default_listeners = spec.default_listeners # "1.1.1.*,2.2.2.*"
+        if default_listeners:
+            listeners_ip = ""
+            for listeners_format in default_listeners.split(','):
+                hosts = [h.hostname for h in spec.placement.hosts]
```

Comment:
are we okay with this limiting the feature to only when using direct host lists for placements? For example if I was using labels this wouldn't work. Actually I think you might end up getting a `'NoneType' object is not iterable` error if someone tried which we should do something about.

Reply:
That kind of surprises me, but if it works, it works. Would you mind adding something to this PR to include a teuthology test that includes setting those fields (or an update to https://github.com/ceph/ceph/blob/main/qa/suites/orch/cephadm/smoke-roleless/2-services/nvmeof.yaml, which is quite simplistic at the moment) just so we can make sure that's getting tested?